packages/envd/internal/services/filesystem/stat.go

```diff
@@ -31,12 +32,21 @@ func (Service) Stat(ctx context.Context, req *connect.Request[rpc.StatRequest])
 		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("error statting file: %w", err))
 	}
 
+	owner, group := getFileOwnership(fileInfo)
+	fileMode := fileInfo.Mode()
+
 	return connect.NewResponse(
 		&rpc.StatResponse{
 			Entry: &rpc.EntryInfo{
```

Comment:
could you simplify by some util function which would create `EntryInfo` from `fileInfo`? So you don't have to double the logic?